crates/wasmtime/src/runtime.rs

```diff
@@ -730,8 +732,23 @@ pub struct Store {
 pub(crate) struct StoreInner {
     engine: Engine,
     compiler: RefCell<Compiler>,
+    interrupts: Arc<VMInterrupts>,
+    signatures: RefCell<SignatureRegistry>,
     instances: RefCell<Vec<InstanceHandle>>,
     signal_handler: RefCell<Option<Box<SignalHandler<'static>>>>,
+    jit_code_ranges: RefCell<Vec<(usize, usize)>>,
+}
+
+pub(crate) struct JitCodeRegistration {
```

Comment:
I think in general this probably isn't necessary if we do this registration at instantiation time instead of compile time for a `Module`. Once a `Module` is instantiated into a `Store` we never drop the `InstanceHandle` anyway, so it's permanently registered within a store, which means there's no need to remove.